bids/variables/variables.py

```diff
@@ -447,18 +447,33 @@ def resample(self, sampling_rate, inplace=False, kind='linear'):
 
         self.index = self._build_entity_index(self.run_info, sampling_rate)
 
-        x = np.arange(n)
         num = len(self.index)
 
-        from scipy.interpolate import interp1d
-        f = interp1d(x, self.values.values.ravel(), kind=kind)
-        x_new = np.linspace(0, n - 1, num=num)
-        self.values = pd.DataFrame(f(x_new))
+        if integration_window is not None:
+            # Number of timesteps in existing time series
+            n_frames = int(np.floor(integration_window * old_sr))
+            old_times = np.arange(n) / old_sr
+            new_times = np.arange(num) / sampling_rate
+            diffs = old_times.reshape((1, -1)) - new_times.reshape((-1, 1))
+            integrator = ((diffs >= 0) & (diffs < integration_window)).astype(np.uint8)
```

Comment:
i think this could be represented as a sparse matrix times vector operation and be efficient in memory. it's effectively a boxcar per sample. [0000 11111111 0000000]*values

Reply:
+1. I'm sure there are all kinds of optimizations we can apply here. We could also try JIT-ing this approach—I'm not at all averse to adding an (optional) numba dependency, and there are other places in the `Transformation` module where numba would probably help.